### PR TITLE
Work part 2

### PR DESCRIPTION
**Pull request by anas14680**
This pull request adds a new functionality to ehr library and also tests them. Some errors found in code after testing have also been corrected. @patrickkwang

### CODE REVIEW

File: Bio_stats_assignment.py [hunk 1]
```diff
@@ -141,3 +142,68 @@ def sick_patients(
     return unique_patients_  # O(1)
 
     # complexity of the above function is O(2+1+1+N+N+N+1) -> O(5+3N) -> O(N)
+
+
+def patient_age(
```
Comment (patrickkwang) [inline Bio_stats_assignment.py]: I appreciate the thorough comments. The function name could be more informative. In my mind, "patient age" is just their current age...

File: tests.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            "1996-04-04 02:45:40.547",
+            "1976-12-28 02:45:40.547",
+            "1984-02-28 02:45:40.547",
+            "1990-07-28 02:45:40.547",
+        ],
+        "PatientRace": ["Asian", "White", "Asian", "White"],
+    }
+
+    # test a sample lab test data
+    labs_data_test = "/Users/mohammadanas/Desktop/Duke MIDS/Spring 2021\
```
Comment (patrickkwang) [inline tests.py]: If you use paths for your machine, no one who clones this repo can run the code. Instead, commit the test data files and make these relative paths.

**Review by patrickkwang — CHANGES_REQUESTED** [review]
Nice job! Minor comments.

File: tests.py [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+    # 25, 45, 34, 32
+
+    assert num_older_than(26, patient_data) == 3
+    assert (
+        num_older_than(25, patient_data) == 4
+    )  # checks if age integer is greater than actual age like
+    # person who is aged 25 is technically older than 25. Becasue he
+    # 25 months X months
+    assert num_older_than(38, data_columns) == "No information found regarding Age"
```
Comment (patrickkwang) [inline tests.py]: Should this be returning a string? Or raising an error?

File: tests.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    patient_test_data = "Patient test.txt"
+    patient_data = parse_data(patient_test_data)
+
+    # a basic test to see if the function works correctly
+    assert patient_age_on_first_test("ANAS123", parsed_lab_data, patient_data) == 25
+
+    # To check if the function works fine if patient not found in data
+    assert (
+        patient_age_on_first_test("ANNA123", parsed_lab_data, patient_data)
+        == "No information available on patient with ID:ANNA123"
```
Comment (patrickkwang) [inline tests.py]: Should this be returning a string? Or raising an error?

**Review by patrickkwang — CHANGES_REQUESTED** [review]

File: Bio_stats_assignment.py [hunk 5]
```diff
@@ -62,7 +62,7 @@ def num_older_than(
     today_date: datetime = datetime.today()  # O(1) generate todays date
 
     if "PatientDateOfBirth" not in patient_data.keys():  # O(1)
-        return "No information found regarding Age"  # O(1)
+        raise KeyError(f"No information found regarding Age")  # O(1)
```
Comment (anas14680) [inline Bio_stats_assignment.py:65]: @patrickkwang  Professor I have changed this from returning strings to raising errors 
Please let me know if there are any other changes required 
Thank you

Comment (anas14680) [reply Bio_stats_assignment.py:65]: These changes have been made through out the python script (in other places ) and the test file

**Review by patrickkwang — APPROVED** [review]

**Pull request merged**